linux/x86-64/python3/acrcloud/async_recognizer.py

```diff
@@ -0,0 +1,278 @@
+            try:
+                json.loads(res)
+            except Exception as e:
+                res = ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.JSON_ERROR_CODE, str(res))
+        except Exception as e:
+            res = ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.UNKNOW_ERROR_CODE, str(e))
+        return res
+
+    async def recognize_by_file(self, file_path, start_seconds, rec_length=10):
+        res = ''
+        try:
+            query_data = {}
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_AUDIO or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample'] = acrcloud_extr_tool.create_fingerprint_by_file(
+                    file_path, start_seconds, rec_length, False)
+
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_HUMMING or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample_hum'] = acrcloud_extr_tool.create_humming_fingerprint_by_file(file_path,
+                                                                                                 start_seconds,
+                                                                                                 rec_length)
+
+            res = await self.do_recogize(self.host, query_data, self.query_type, self.access_key, self.access_secret,
+                                   self.timeout)
+            try:
+                json.loads(res)
+            except Exception as e:
+                res = ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.JSON_ERROR_CODE, str(res))
+        except Exception as e:
+            res = ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.UNKNOW_ERROR_CODE, str(e))
+        return res
+
+    async def recognize_by_filebuffer(self, file_buffer, start_seconds, rec_length=10):
+        res = ''
+        try:
+            query_data = {}
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_AUDIO or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample'] = acrcloud_extr_tool.create_fingerprint_by_filebuffer(file_buffer, start_seconds,
+                                                                                           rec_length, False)
+
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_HUMMING or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample_hum'] = acrcloud_extr_tool.create_humming_fingerprint_by_filebuffer(file_buffer,
+                                                                                                       start_seconds,
+                                                                                                       rec_length)
+
+            res = await self.do_recogize(self.host, query_data, self.query_type, self.access_key, self.access_secret,
+                                   self.timeout)
```

Comment:
And yet more, here and the lines above.  Let's minimize the differences since diff is going to be our maintenance method for at least the short term.